fabric-structure-api-v1/src/main/java/net/fabricmc/fabric/api/structure/v1/FabricStructurePool.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.structure.v1;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.mojang.datafixers.util.Pair;
+
+import net.minecraft.structure.pool.StructurePool;
+import net.minecraft.structure.pool.StructurePoolElement;
+
+import net.fabricmc.fabric.mixin.structure.StructurePoolAccessor;
+
+/*
+ * Represents a modifiable structure pool that would have several helper methods for modders.
+ */
+public class FabricStructurePool {
+	private final StructurePool pool;
+
+	public FabricStructurePool(StructurePool underlying) {
+		this.pool = underlying;
+	}
+
+	public void addStructurePoolElement(StructurePoolElement element) {
+		addStructurePoolElement(element, 1);
+	}
+
+	public void addStructurePoolElement(StructurePoolElement element, int weight) {
+		//adds to elementCounts list
+		List<Pair<StructurePoolElement, Integer>> list = new ArrayList<>(((StructurePoolAccessor) pool).getElementCounts());
+		list.add(Pair.of(element, weight));
+		((StructurePoolAccessor) pool).setElementCounts(list);
+
+		//adds to elements list
+		for (int i = 0; i < weight; i++) {
+			((StructurePoolAccessor) pool).getElements().add(element);
+		}
+	}
+
+	public StructurePool getStructurePool() {
```

Comment:
Odd naming, you don't really get a structure pool from a (Fabric) structure pool.

Reply:
should it be getUnderlying and make the field name "underlying"?

Reply:
You can probably make `FabricStructurePool` an interface that's injected by mixin onto regular structure pools, like https://github.com/FabricMC/fabric/blob/351679a7decdd3044d778e74001de67463bee205/fabric-loot-tables-v1/src/main/java/net/fabricmc/fabric/api/loot/v1/FabricLootPool.java#L31